libraries/botframework-schema/src/teams/index.ts

```diff
@@ -1225,6 +1229,167 @@ export interface FileConsentCardResponse {
+ * Current TabRequest entity context, or 'tabEntityId'.
+ *
+ */
+export interface TabEntityContext {
+    /**
+     * @member {string} [tabEntityId] The entity id of the tab.
+     */
+    tabEntityId?: string;
+}
+
+/**
+ * @interface
+ * Invoke ('tab/fetch') request value payload.
+ *
+ */
+export interface TabRequest {
+    /**
+     * @member {TabEntityContext} [tabContext] The current tab entity request context.
+     */
+    tabContext?: TabEntityContext;
+    /**
+     * @member {TabContext} [context] The current user context, i.e., the current theme.
+     */
+    context?: TabContext;
+    /**
+     * @member {string} [state] The magic code for OAuth flow.
+     */
+    state?: string;
+}
+
+/**
+ * @interface
+ * Envelope for Card Tab Response Payload.
+ *
+ */
+export interface TabResponse {
+    /**
+     * @member {TabResponsePayload} [tab] The response to the tab/fetch message.
+     */
+    tab?: TabResponsePayload;
+}
+
+/**
+ * @interface
+ * Envelope for cards for a Tab request.
+ *
+ */
+export interface TabResponseCard {
+    /**
+     * @member {object} [card] The adaptive card for this card tab response.
+     */
+    card?: Record<string, unknown>;
+}
+
+/**
+ * @interface
+ * Envelope for cards for a TabResponse.
+ *
+ */
+export interface TabResponseCards {
+    /**
+     * @member {object[]} [cards] Adaptive cards for this card tab response.
+     */
+    cards?: Record<string, unknown>[];
+}
+
+/**
+ * @interface
+ * Payload for Tab Response.
+ *
+ */
+export interface TabResponsePayload {
+    /**
+     * @member {'continue' | 'auth' | 'silentAuth'} [type] Choice of action options when responding to the tab/fetch message.
+     */
+    type?: 'continue' | 'auth' | 'silentAuth';
+    /**
+     * @member {TabResponseCards} [value] The TabResponseCards to send when responding to
+     * tab/fetch activity with type of 'continue'.
+     */
+    value?: TabResponseCards;
+    /**
+     * @member {TabSuggestedActions} [suggestedActions] The Suggested Actions for this card tab.
+     */
+    suggestedActions?: TabSuggestedActions;
+}
+
+/**
+ * @interface
+ * Invoke ('tab/submit') request value payload.
+ *
+ */
+export interface TabSubmit {
+    /**
+     * @member {TabEntityContext} [tabContext] The current tab's entity request context.
+     */
+    tabContext?: TabEntityContext;
+    /**
+     * @member {TabContext} [context] The current user context, i.e., the current theme.
+     */
+    context?: TabContext;
+    /**
+     * @member {TabSubmitData} [data] User input. Free payload containing properties of key-value pairs.
+     */
+    data?: TabSubmitData;
+}
+
+/**
+ * @interface
+ * Invoke ('tab/submit') request value payload data.
+ *
+ */
+export interface TabSubmitData {
+    /**
+     * @member {string} [type] Should currently be 'tab/submit'.
+     */
+    type?: string;
+    /**
+     * @member {any} [any] Additional properties not otherwise defined in TabSubmit.
+     */
+    [properties: string]: any;
+}
+
```

Comment:
This was another area where I wasn't sure how to match with JSDoc, particularly around `any` (which other interfaces in this file use) and `unknown`, which isn't recognized by JSDoc if I use `{uknown}`.

Reply:
For JSDoc, I use `any` in place of `unknown`. The `unknown` modifier is more of a compiler nudge as you can use any type to satisfy `unknown`. It just forces you to type check if you access any type-specific methods or operations.

Reply:
okay as-is then, using `any` for both the JSDoc comments and the `[properties: string]: any;` line?

Reply:
Leave it as is, `unknown` should be the _actual_ type, `any` is simply the most appropriate JSDoc annotation.